LaTeX/LaTeX.sublime-syntax

```diff
@@ -198,6 +207,123 @@ contexts:
+    - include: brace-group-end
+    - include: macro-braces
+    - include: xparse-single-argspec
+    # not actually allowed, but included here to future-proof highlighting
+    - include: general-constants
+    - include: general-commands
+    - include: paragraph-pop
+
+  xparse-single-argspec:
+    # modifiers and simple specs
+    - match: '[+!movsb]'
+      scope: string.unquoted.latex meta.annotation.parameters.latex
+
+    - match: t
+      scope: string.unquoted.latex meta.annotation.parameters.latex
+      push:
+        - xparse-args-token
+
+    # delimited arguments without default
+    - match: '[rd]'
+      scope: string.unquoted.latex meta.annotation.parameters.latex
+      push:
+        - xparse-args-token
+        - xparse-args-token
+
+    # delimited arguments with default
+    - match: '[RD]'
+      scope: string.unquoted.latex meta.annotation.parameters.latex
+      push:
+        - xparse-optarg-value
+        - xparse-args-token
+        - xparse-args-token
+
+    # optional argument with default
+    - match: O
+      scope: string.unquoted.latex meta.annotation.parameters.latex
+      push: xparse-optarg-value
+
+    # embellishment argument
+    - match: e
+      scope: string.unquoted.latex meta.annotation.parameters.latex
+      push: xparse-embellishment
+
+    # embellishment argument with default
+    - match: E
+      scope: string.unquoted.latex meta.annotation.parameters.latex
+      push:
+        - xparse-optarg-value
+        - xparse-embellishment
+
+  xparse-embellishment:
+    - match: \{
+      scope: punctuation.definition.group.brace.begin.latex
+      set: xparse-embellishment-content
+    - match: (?=\S)
+      set: xparse-args-token
+    - include: paragraph-pop
+
+  xparse-embellishment-content:
+    - match: (?!\})
+      push: xparse-args-token
+    - include: brace-group-end
+    - include: paragraph-pop
+    - include: else-pop
+
+  xparse-args-token:
+    # either a command
+    - match: '(\\){{cmdname}}{{endcs}}'
+      scope: constant.other.token.latex
+      captures:
+        1: punctuation.definition.backslash.latex
+      pop: 1
+    # or the token wrapped in a group
+    - match: \{
+      scope: punctuation.definition.group.brace.begin.tex
+      push: xparse-args-token
+    - match: \}
+      scope: punctuation.definition.group.brace.end.tex
+      pop: 1
+    # or a single character
+    - match: '\S'
+      scope: constant.character.latex constant.other.token.latex
+      pop: 1
+    - include: paragraph-pop
+    - include: else-pop
+
+  xparse-optarg-value:
+    - match: '\{'
+      scope: punctuation.definition.group.brace.begin.latex
+      push:
+        - meta_scope: meta.function.parameters.default-value.latex
+        - match: '\}'
+          scope: punctuation.definition.group.brace.end.latex
+          pop: 2
+        - include: general-constants
+        - include: general-commands
+        - include: macro-braces
+    - match: (?=\})
+      pop: 1
```

Comment:
Maybe

1. use a named context
2. use a more generic bailout (`else-pop`) instead of `(?=\})`?
3. maybe add a `clear_scope: 1` to avoid stacking multiple `meta.function.parameter` scopes? _(not part of my suggestion)_
```suggestion
    - match: \{
      scope: punctuation.definition.group.brace.begin.latex
      push: xparse-optarg-value-body
    - include: else-pop

  xparse-optarg-value-body:
    - meta_scope: meta.function.parameters.default-value.latex
    - match: \}
      scope: punctuation.definition.group.brace.end.latex
      pop: 2
    - include: general-constants
    - include: general-commands
    - include: macro-braces
```

Reply:
I think I've fixed this up now, with better scoping for the case if there are no {} and avoiding double meta-scopes